bin/spack-modify-makefile-push-buildcache

```diff
@@ -3,14 +3,41 @@
 import fileinput
 import re
 import sys
+import subprocess
 
+# Spack can either install packages from a build cache or to compile them.
```

Comment:
This is not entirely true - spack could install from cache and fallback to compiling BUT for what we want:
- Try installing from buildcache and if successful, then do not push anything to buildcache (obviously it was there already)
- If installing from cache failed, try installing from source
- If installing from source is successful try pushing to cache
- If pushing to cache fails, this should be non-fatal
This is somewhat difficult to hide in a flag in spack itself, because it is not obvious that pushing to cache is NOT an error.

The problem we are facing are in fact 2 problems:
- If we let spack to choose to install from spack OR source, and a faulty buildcache artifact is in cache,then spack will fail and not continue to build from source (maybe that's a bug in spack - not sure if this should be fixed on the spack side). It would only fallback to install from source if no artifact was found in the first place, but never when it was found but broken.
- Assuming the previous problem would be "fixed" in spack, it is not clear whether the artifact was built from cache or source, and we only want to push to cache if it was installed from source.

I think what I want is this flag: `spack --install-from-cache-and always-fallback-to-install-from-source-with-subsequent-nonfatal-push-to-cache` :sweat:

Reply:
TODO: Update https://github.com/finkandreas/spack